CRM/Utils/Cache/NaiveMultipleTrait.php

```diff
@@ -92,11 +99,23 @@ public function setMultiple($values, $ttl = NULL) {
    *   or if any of the $keys are not a legal value.
    */
   public function deleteMultiple($keys) {
+    $this->assertIterable('deleteMultiple', $keys);
+
     $result = TRUE;
     foreach ($keys as $key) {
       $result = $this->delete($key) || $result;
     }
     return $result;
   }
 
+  /**
+   * @param $keys
+   * @throws \CRM_Utils_Cache_InvalidArgumentException
+   */
+  private function assertIterable($func, $keys) {
+    if (!is_array($keys) || $keys instanceof Traversable) {
```

Comment:
I feel like this should be reversed ie

```
While you cannot implement this interface, you can use it in your checks to determine if something is usable in for each. Here is what I use if I'm expecting something that must be iterable via foreach.
in the example not !items
```
<?php
    if( !is_array( $items ) && !$items instanceof Traversable )
        //Throw exception here
?>
```
http://php.net/manual/en/class.traversable.php

Reply:
Oops, good catch! This explains why `testGetMultipleWithGenerator()`,`testSetMultipleWithGenerator()`, and `testDeleteMultipleGenerator()` weren't working. :)

I've pushed up a patch to fix `assertIterable` and re-enable those tests.